tools/cmd/seidb/dump_db.go

```diff
@@ -0,0 +1,91 @@
+package main
+
+import (
+	"fmt"
+	"io/fs"
+	"os"
+
+	"github.com/sei-protocol/sei-db/ss"
+	"github.com/sei-protocol/sei-db/tools/utils"
+	"github.com/spf13/cobra"
+)
+
+const outputFileName = "db_dump.kv"
+
+func DumpDbCmd() *cobra.Command {
+	dumpDbCmd := &cobra.Command{
+		Use:   "dump-db",
+		Short: "For a given State Store DB, dump-db iterates over all keys and values for a specific store and writes them to a file",
+		Run:   dump,
+	}
+
+	dumpDbCmd.PersistentFlags().StringP("output-dir", "o", "", "Output Directory")
+	dumpDbCmd.PersistentFlags().StringP("db-dir", "d", "", "Database Directory")
+	// TODO: Accept multiple modules. Can pass empty to iterate over all stores
+	dumpDbCmd.PersistentFlags().StringP("module", "m", "", "Module to export. Leave empty to export all")
+	dumpDbCmd.PersistentFlags().StringP("db-backend", "b", "", "DB Backend")
+
+	return dumpDbCmd
+}
+
+func dump(cmd *cobra.Command, _ []string) {
+	outputDir, _ := cmd.Flags().GetString("output-dir")
+	module, _ := cmd.Flags().GetString("module")
+	dbDir, _ := cmd.Flags().GetString("db-dir")
+	dbBackend, _ := cmd.Flags().GetString("db-backend")
+
+	if dbDir == "" {
+		panic("Must provide database dir")
+	}
+
+	if dbBackend == "" {
+		panic("Must provide db backend")
+	}
+
+	_, isAcceptedBackend := ValidDBBackends[dbBackend]
+	if !isAcceptedBackend {
+		panic(fmt.Sprintf("Unsupported db backend: %s\n", dbBackend))
+	}
+
+	if outputDir == "" {
+		panic("Must provide output dir when generating db export")
+	}
+
+	DumpDbData(dbBackend, module, outputDir, dbDir)
+}
+
+// Outputs the raw keys and values for all modules at a height to a file
+func DumpDbData(dbBackend string, module string, outputDir string, dbDir string) {
+	// Create output directory
+	err := os.MkdirAll(outputDir, fs.ModePerm)
+	if err != nil {
+		panic(err)
+	}
+
+	// Create output directory
+	currentFile, err := utils.CreateFile(outputDir, fmt.Sprintf(outputFileName))
+	if err != nil {
+		panic(err)
+	}
+	defer currentFile.Close()
+
+	// TODO: Defer Close Db
+	backend, err := ss.NewStateStoreDB(dbDir, ss.BackendType(dbBackend))
+	if err != nil {
+		panic(err)
+	}
+
+	fmt.Printf("Writing db data to %s...\n", outputFileName)
+
+	// Callback to write db entries to file
+	_, err = backend.RawIterate(module, func(key, value []byte, version int64) bool {
+		_, err = currentFile.WriteString(fmt.Sprintf("Key: %X Val: %X Version: %d\n", key, value, version))
+		if err != nil {
+			panic(err)
```

Comment:
Likely cleaner way to handle this instead of panic, but iterator will be closed properly